File: frontend/src/services/authService.js
import api from './api';

export const login = (data) => {
  return api.post('/auth/login', data);
};

export const register = (data) => {
  return api.post('/auth/register', data);
};

export const verifyRegistrationOtp = (email, otp) => {
  return api.post('/auth/verify-registration-otp', { email, otp });
};

export const resendRegistrationOtp = (email) => {
  return api.post('/auth/resend-registration-otp', { email });
};

export const logout = (token) => {
  return api.post(
    '/auth/logout',
    {},
    {
      headers: token ? { Authorization: `Bearer ${token}` } : {},
    }
  );
};

export const forgotPassword = (email) => {
  return api.post('/auth/forgot-password', { email });
};

export const resetPassword = (data) => {
  return api.post('/auth/reset-password', data);
};

export const refreshToken = (token) => {
  return api.post('/auth/refresh-token', { refreshToken: token });
};

File: backend/src/main/java/com/handmade/store/service/AuthService.java
package com.handmade.store.service;

import com.handmade.store.dto.auth.AuthResponse;
import com.handmade.store.dto.auth.ForgotPasswordRequest;
import com.handmade.store.dto.auth.LoginRequest;
import com.handmade.store.dto.auth.RegisterRequest;
import com.handmade.store.dto.auth.ResetPasswordRequest;

import java.util.Map;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    Map<String, Object> forgotPassword(ForgotPasswordRequest request);

    Map<String, Object> resetPassword(ResetPasswordRequest request);

    AuthResponse refreshToken(String refreshToken);

    Map<String, Object> verifyRegistrationOtp(String email, String otp);

    Map<String, Object> resendRegistrationOtp(String email);

    Map<String, Object> logout(String token);
}
